rollout/experiment_test.go

```diff
@@ -820,3 +820,48 @@ func TestRolloutCreateExperimentWithService(t *testing.T) {
 	assert.Equal(t, "canary-template", ex.Spec.Templates[1].Name)
 	assert.Nil(t, ex.Spec.Templates[1].Service)
 }
+
+// TestRolloutCreateWeightlessExperimentWithService does the same as TestRolloutCreateExperimentWithService, but when weight is not set.
+// CreateService is true when Service is set, even when Weight isn't, otherwise false.
+func TestRolloutCreateWeightlessExperimentWithService(t *testing.T) {
+	steps := []v1alpha1.CanaryStep{{
+		Experiment: &v1alpha1.RolloutExperimentStep{
+			Templates: []v1alpha1.RolloutExperimentTemplate{
+				// Service should be created for "stable-weightless-template"
+				{
+					Name:     "stable-weightless-template",
+					SpecRef:  v1alpha1.StableSpecRef,
+					Replicas: pointer.Int32Ptr(1),
+					Service: &v1alpha1.TemplateService{
+						Name: "test-service",
```

Comment:
please add test without name (if it exists, please ignore comment)

Reply:
Done. See: https://github.com/argoproj/argo-rollouts/pull/2633/files#diff-4ee6ea6eacb6c4ad54a87a51cdba981669b25bbd2d30a0b51bbd8fcd56bce155R870